dashboard/app/models/circuit_playground_discount_application.rb

```diff
@@ -0,0 +1,73 @@
+# == Schema Information
+#
+# Table name: circuit_playground_discount_applications
+#
+#  id                                  :integer          not null, primary key
+#  user_id                             :integer          not null
+#  unit_6_intention                    :integer
+#  has_confirmed_school                :boolean          default(FALSE), not null
+#  full_discount                       :boolean
+#  admin_set_status                    :boolean          default(FALSE), not null
+#  signature                           :string(255)
+#  signed_at                           :datetime
+#  circuit_playground_discount_code_id :integer
+#  created_at                          :datetime         not null
+#  updated_at                          :datetime         not null
+#
+# Indexes
+#
+#  index_circuit_playground_applications_on_code_id           (circuit_playground_discount_code_id)
+#  index_circuit_playground_discount_applications_on_user_id  (user_id)
+#
+
+class CircuitPlaygroundDiscountApplication < ApplicationRecord
+  belongs_to :user
+  belongs_to :circuit_playground_discount_code
```

Comment:
Rails question: What does it mean for record validation that we have a `belongs_to` relation here but we expect `circuit_playground_discount_code_id` to often be null?

Reply:
That's fine. Whenever the id is null, the association will also be null.